### ISSUE TITLE
How to display the ACC for each category on the console

### ISSUE THREAD

**Issue by SweetyXt** [ultralytics/ultralytics#8982]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

During my verification process for image classification, the console only shows the overall ACC, but there is no single category, and I have a total of 8 categories
This is the output of my console：
 classes   top1_acc   top5_acc: 100%|██████████| 120/120 [06:10<00:00,  3.09s/it]
                   all          1          1
Speed: 0.0ms preprocess, 118.6ms inference, 0.0ms loss, 0.0ms postprocess per image
Results saved to runs/classify/val

### Additional

help please

**Comment by glenn-jocher** [comment]
@SweetyXt hey there! 🙌 To display category-wise accuracy on the console during your validation process, you could modify the validation code to compute and print the accuracy for each category after all images have been processed.

Assuming you're working with YOLOv8 Python API, after the validation loop, you have the predictions and true labels for each image. You can then calculate the accuracy per category like so:

```python
from sklearn.metrics import accuracy_score

# Assuming `preds` and `labels` contain your predictions and true labels respectively
# And `num_classes` is the total number of categories (8 in your case)

accuracies = []
for i in range(num_classes):
    # Filter predictions and labels for current class
    mask = labels == i
    class_preds = preds[mask]
    class_labels = labels[mask]
    
    # Calculate accuracy for current class
    acc = accuracy_score(class_labels, class_preds)
    accuracies.append(acc)
    
    print(f'Class {i} Accuracy: {acc*100:.2f}%')

print("Overall ACC:", np.mean(accuracies) * 100)
```

This snippet uses `accuracy_score` from `sklearn.metrics` but you can adapt the calculation method to your needs if sklearn isn't preferred. This assumes `preds` and `labels` are numpy arrays of your model predictions and true labels for each image, respectively. 

Keep in mind this is a general approach to guide you, and specific modifications might be necessary based on how your prediction loop is structured. This way, you'd get an accuracy printout for each category directly on your console. Happy coding! 😊
